index.rst

```diff
@@ -5,77 +5,331 @@
-The producer would then read the alert information from the EFD and display it.
+Combining rules would be something like:
+
+.. code-block:: yaml
+
+   rule:
+      monitorEvent:
+         component: ATCamera
+         event: endReadout
+      compareTopics:
+         topic1: ATMCS.tel_mount_AzEl_Encoders.azimuthCalculatedAngle
+         topic2: WeatherStation.windDirection.value
+         threshold: 1.0
+
+The advantage of this approach is that, once a sufficient number of rules are developed, including new triggers is a matter of creating a new configuration.
+
+Alternatively, the programmatic-based rules relies on having all the trigger logic developed under a single operation.
+For example, we could have a rule to monitor the end readout event from the camera and a rule that monitors the end readout event from the camera and compare the position of the telescope with the wind.
+These rules are developed specifically for their use-cases, and are not generally applied.
+This is, for instance, the current approach implemented by the ``Watcher``.
+Any new alarm requires code changes. 
+
+Analysis Process
+----------------
+
+A key functionality of the ``Catcher``, that differentiates it from the ``Watcher``, is the ability to execute an analysis process once a certain condition is met.
+In general, we assume that these analysis will consist of lightweight computations.
+Whenever possible, the computation should be offloaded to the OCPS.
+It is likely that the data processed by the rule will be relevant to the downstream processing, so it should  be made available to it as well.
+
+These processing routines should be developed in the ``Catcher`` codebase using a common framework, similar to that provided by SAL Scripts (e.g. a base class with a well defined interface).
+It is also important that the results/outputs of a process follow a well defined format that can be consumed downstream.
+
+.. _introduction-process-image-data:
 
 Process Image Data
 ------------------
 
-Another important feature of the Catcher was the capability to process camera image data using the OCPS.
-The OCPS is a CSC that is capable of executing DM-pipeline processes.
+Another important feature of the ``Catcher`` is the capability to process camera image data using the ``OCPS``.
+The ``OCPS`` is a CSC that is capable of executing DM-pipeline processes.
 It is going to be key in allowing remote processing of Main Camera data during commissioning and operations.
 
-Nevertheless, if the Catcher is not a CSC it won't be able to interact direcly with OCPS to process image data.
-One alternative would be to interact with the OCPS remote backend direcly, instead of going through the CSC.
-For that, the OCPS backend implements a REST API that we could leverage to post requests and receive responses back.
+Since the ``Catcher`` is a CSC it can interact directly with ``OCPS`` to process image data, by sending commands to it.
+
```

Comment:
Maybe not worth adding, but we expect to use the Catcher when it doesn't make sense to use the Rapid Analysis functionality, which gets run on every image.

Reply:
I still don't know what these eyes mean! :-)